src/main/java/org/broadinstitute/dsm/model/BSPKit.java

```diff
@@ -0,0 +1,150 @@
+package org.broadinstitute.dsm.model;
+
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.InstanceSettings;
+import org.broadinstitute.dsm.db.dao.ddp.kitrequest.KitRequestDao;
+import org.broadinstitute.dsm.db.dto.settings.InstanceSettingsDto;
+import org.broadinstitute.dsm.model.bsp.BSPKitInfo;
+import org.broadinstitute.dsm.model.bsp.BSPKitQueryResult;
+import org.broadinstitute.dsm.model.bsp.BSPKitStatus;
+import org.broadinstitute.dsm.statics.ApplicationConfigConstants;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.dsm.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.util.HashMap;
+import java.util.Map;
+
+public class BSPKit {
+    private static Logger logger = LoggerFactory.getLogger(BSPKit.class);
+
+    public BSPKitStatus getKitStatus(@NonNull String kitLabel, NotificationUtil notificationUtil) {
+        logger.info("Checking label " + kitLabel);
+        BSPKitQueryResult bspKitQueryResult = BSPKitQueryResult.getBSPKitQueryResult(kitLabel);
+
+        if (bspKitQueryResult == null) {
+            logger.info("No kit w/ label " + kitLabel + " found");
+            return null;
+        }
+
+        if (StringUtils.isNotBlank(bspKitQueryResult.getParticipantExitId())) {
+            String message = "Kit of exited participant " + bspKitQueryResult.getBspParticipantId() + " was received by GP.<br>";
+            notificationUtil.sentNotification(bspKitQueryResult.getNotificationRecipient(), message, NotificationUtil.DSM_SUBJECT);
+            return new BSPKitStatus(BSPKitStatus.EXITED);
+        }
+        if (StringUtils.isNotBlank(bspKitQueryResult.getDeactivationDate())) {
+            return new BSPKitStatus(BSPKitStatus.DEACTIVATED);
+        }
+        return null;
```

Comment:
main intent why Java introduced Optional from version 8 is to handle NPEs, main usage of optional is the methods where there is a chance of return of null reference. Additionally Optional not only provides handle of NPEs, with Optional code becomes much more cleaner since you don't have to write all those `null` reference if checks.